telegram/src/main/kotlin/me/ivmg/telegram/network/ApiClient.kt

```diff
@@ -258,6 +260,28 @@ class ApiClient(
         )
     }
 
+    fun sendDocument(
+        chatId: Long,
+        fileBytes: ByteArray,
+        caption: String?,
+        parseMode: String?,
+        disableNotification: Boolean?,
+        replyToMessageId: Long?,
+        replyMarkup: ReplyMarkup?,
+        fileName: String?
+    ): Call<Response<Message>> {
+
+        return service.sendDocument(
+            convertString(chatId.toString()),
+            convertBytes("document", fileBytes, null, fileName ?: caption),
+            if (caption != null) convertString(caption) else null,
+            if (parseMode != null) convertString(parseMode) else null,
+            if (disableNotification != null) convertString(disableNotification.toString()) else null,
+            if (replyToMessageId != null) convertString(replyToMessageId.toString()) else null,
+            if (replyMarkup != null) convertJson(replyMarkup.toString()) else null
+        )
```

Comment:
I would probably change `convertXXX` functions to make them extension functions of `String`, `ByteArray`, etc... to take advantage of Kotlin features. In this case, if `convertString` would have been an extension function the code involving nullables could be easier to read:

```
return service.sendDocument(
    chatId.toString().convertString(),
    ....,
    caption?.convertString(),
    parseMode?.convertString(),
    ...
```

However, that's probably out of scope of this PR, so you can keep it as it is right now and we can consider adding the mentioned functions in a future task.

Reply:
That's fine, but it also can be done by using `?.let{}` like this `caption?.let { convertString(it) }`

Reply:
Sure, I prefer the extension function option because is more readable IMHO but it's probably a matter of taste. Anyways, these changes are out of the scope of the PR you did, maybe we can discuss them for a future task.